app/src/main/java/org/simple/clinic/summary/PatientSummaryEffectHandler.kt

```diff
@@ -224,6 +225,23 @@ class PatientSummaryEffectHandler @AssistedInject constructor(
     }
   }
 
+  private fun loadDataForBackClick(
+      scheduler: Scheduler
+  ): ObservableTransformer<LoadDataForBackClick, PatientSummaryEvent> {
+    return ObservableTransformer { effects ->
+      effects
+          .observeOn(scheduler)
+          .map { loadDataForBackClick ->
+            val patientUuid = loadDataForBackClick.patientUuid
+            val timestamp = loadDataForBackClick.screenCreatedTimestamp
+            DataForBackClickLoaded(
+                hasPatientDataChangedSinceScreenCreated = patientRepository.hasPatientDataChangedSince(patientUuid, timestamp),
+                noBloodPressuresRecordedForPatient = doesNotHaveBloodPressures(patientUuid)
```

Comment:
1. We cannot make `hasPatientDataChangedSinceScreenCreated` part of the model because it involves the database.
2. I initially thought of making them a part of the model, but then I realised that they should not be there.
  - If we look at how these values are being used in `PatientSummaryUpdate`, they never lead to a change in the screen state because they always result in a `dispatch` and not in a `next`. They are never used in the `ViewRenderer`.
  - We use them to decide navigation which should not be handled as part of the regular screen flow and should be handled elsewhere. The only reason we have them here is because we are not going to make too many changes at once (limiting ourselves to breaking out a separate `UiActions` only).

Does this make sense?

Reply:
>     1. We cannot make `hasPatientDataChangedSinceScreenCreated` part of the model because it involves the database.
> 
This is why I was recommending to move this logic to the `Model` so that we don't put this in the database. Also, this would mean that we don't use `Effects` to handle navigation either.
>     2. I initially thought of making them a part of the model, but then I realised that they should not be there.
> 
> 
>     * If we look at how these values are being used in `PatientSummaryUpdate`, they never lead to a change in the screen state because they always result in a `dispatch` and not in a `next`. They are never used in the `ViewRenderer`.
> 
That's right and that is the point I am making. Because we rely on a database call to decide whether user should navigate to `ScheduleAppointmentSheet` or to exit, we end up dispatching an effect. I am suggesting that we use `UiRenderer` for this change by making it part of the `Model`.
>     * We use them to decide navigation which should not be handled as part of the regular screen flow and should be handled elsewhere. The only reason we have them here is because we are not going to make too many changes at once (limiting ourselves to breaking out a separate `UiActions` only).
> 
> 
> Does this make sense?
I just wanted to call out that we can utilize Mobius to achieve the navigation logic rather than the database state.
Let's discuss this in person. As of now I do not expect you to change this implementation.

Reply:
The fact that they're being used only for navigation is precisely why they should not be added to the `Model`. The `Model` is meant to record only the data that is needed to render the screen and nothing else.

Let's discuss this in person tomorrow.